spec/models/assignment_spec.rb

```diff
@@ -2227,6 +2227,42 @@ def grouping_count(groupings)
     end
   end
 
+  describe '#get_num_valid' do
+    before :each do
+      @assignment = create(:assignment, assignment_properties_attributes: { group_min: 2, group_max: 3 })
+      @groupings = Array.new(4) { create(:grouping, assignment: @assignment) }
```

Comment:
Use `let!` for this one---see above comment, and also `let!` will ensure all these groupings get created before each test.

Also, instead of manually creating an array, use the [create_list](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#building-or-creating-multiple-records) FactoryBot helper.